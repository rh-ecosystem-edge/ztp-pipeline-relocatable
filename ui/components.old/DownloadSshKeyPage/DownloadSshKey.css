.download-item {
  background-color: #e7f1fa;
  border-radius: 2px; /* keep in sync with Page.css */
}

.download-item .pf-l-grid {
  margin-left: 1rem;
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.download-item__icon {
  border-right: 1px;
  border-right-style: solid;
  padding-right: var(--pf-global--spacer--md);
  border-right-color: var(--pf-global--palette--black-400);
}

.download-item__text {
  margin-left: var(--pf-global--spacer--md);
  color: var(--pf-global--palette--black-600);
}

.download-item__error {
  margin-top: var(--pf-global--spacer--sm);
}
