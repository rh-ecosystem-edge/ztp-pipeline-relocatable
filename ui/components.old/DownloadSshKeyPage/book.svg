<?xml version="1.0" encoding="UTF-8"?>
<svg width="39px" height="50px" viewBox="0 0 39 50" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <title>AB63849D-0862-4E11-9057-7443FC8A302B</title>
    <g id="Concept-1---Approved" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="6.-Private-ssh-key" transform="translate(-827.000000, -614.000000)" fill="#0066CC" fill-rule="nonzero">
            <g id="Modal" transform="translate(480.000000, 220.000000)">
                <g id="Form-Field-Group" transform="translate(309.000000, 278.000000)">
                    <g id="Stacked-Group-2" transform="translate(22.000000, 93.000000)">
                        <g id="Stacked-Group" transform="translate(16.000000, 16.000000)">
                            <g id="Icon-Red_Hat-eBook-A-Black-RGB" transform="translate(0.000000, 7.000000)">
                                <path d="M37.848599,0.000144148988 L1.13831611,0.000144148988 C0.834985228,-0.00471742417 0.542602882,0.113569611 0.328086246,0.328086246 C0.113569611,0.542602882 -0.00471742417,0.834985228 0.000144148988,1.13831611 L0.000144148988,48.8616839 C-0.00471742417,49.1650148 0.113569611,49.4573971 0.328086246,49.6719138 C0.542602882,49.8864304 0.834985228,50.0047174 1.13831611,49.9998559 L37.848599,49.9998559 C38.1519299,50.0047174 38.4443122,49.8864304 38.6588289,49.6719138 C38.8733455,49.4573971 38.9916326,49.1650148 38.986771,48.8616839 L38.986771,1.13831611 C38.9916326,0.834985228 38.8733455,0.542602882 38.6588289,0.328086246 C38.4443122,0.113569611 38.1519299,-0.00471742417 37.848599,0.000144148988 Z M36.7105802,47.7236651 L2.27633488,47.7236651 L2.27633488,2.27633488 L36.7105802,2.27633488 L36.7105802,47.7236651 Z" id="Shape"></path>
                                <path d="M19.0529342,7.70945675 C19.5906686,7.95628478 20.227978,7.75738515 20.5298652,7.24851732 C20.8317524,6.73964949 20.7008202,6.08498846 20.2264367,5.73137477 C19.7520531,5.37776109 19.0872677,5.43928043 18.6858313,5.87394261 C18.2474812,6.32313991 18.2474812,7.03999775 18.6858313,7.48919506 C18.7882285,7.59168111 18.9143184,7.66733501 19.0529342,7.70945675 L19.0529342,7.70945675 Z" id="Path"></path>
                                <path d="M23.1644859,42.2171227 L15.8224293,42.2171227 C15.1939189,42.2171227 14.6844105,42.726631 14.6844105,43.3551415 C14.6844105,43.9836519 15.1939189,44.4931602 15.8224293,44.4931602 L23.1644859,44.4931602 C23.7929962,44.4931602 24.3025045,43.9836518 24.3025045,43.3551415 C24.3025045,42.7266311 23.7929962,42.2171227 23.1644859,42.2171227 L23.1644859,42.2171227 Z" id="Path"></path>
                                <path d="M8.48037269,16.960448 L19.4934576,16.960448 C20.121968,16.960448 20.6314763,16.4509397 20.6314763,15.8224293 C20.6314763,15.1939189 20.121968,14.6844105 19.4934576,14.6844105 L8.48037269,14.6844105 C7.85186228,14.6844105 7.34235392,15.1939189 7.34235392,15.8224293 C7.34235392,16.4509397 7.85186228,16.960448 8.48037269,16.960448 L8.48037269,16.960448 Z" id="Path"></path>
                                <path d="M8.48037269,22.4669905 L30.5065424,22.4669905 C31.1350528,22.4669905 31.6445612,21.9574821 31.6445612,21.3289717 C31.6445612,20.7004613 31.1350528,20.1909529 30.5065424,20.1909529 L8.48037269,20.1909529 C7.85186228,20.1909529 7.34235392,20.7004613 7.34235392,21.3289717 C7.34235392,21.9574821 7.85186228,22.4669905 8.48037269,22.4669905 Z" id="Path"></path>
                                <path d="M8.48037269,27.9735329 L30.5065424,27.9735329 C31.1350528,27.9735329 31.6445611,27.4640245 31.6445611,26.8355141 C31.6445611,26.2070038 31.1350528,25.6974954 30.5065424,25.6974954 L8.48037269,25.6974954 C7.85186231,25.6974954 7.34235399,26.2070038 7.34235399,26.8355141 C7.34235399,27.4640245 7.85186231,27.9735329 8.48037269,27.9735329 Z" id="Path"></path>
                                <path d="M8.48037269,33.4800754 L30.5065424,33.4800754 C31.1350528,33.4800754 31.6445612,32.970567 31.6445612,32.3420566 C31.6445612,31.7135462 31.1350528,31.2040378 30.5065424,31.2040378 L8.48037269,31.2040378 C8.07379772,31.2040378 7.69810694,31.4209429 7.49481945,31.7730472 C7.29153195,32.1251514 7.29153195,32.5589617 7.49481945,32.911066 C7.69810694,33.2631702 8.07379772,33.4800754 8.48037269,33.4800754 L8.48037269,33.4800754 Z" id="Path"></path>
                            </g>
                        </g>
                    </g>
                </g>
            </g>
        </g>
    </g>
</svg>
