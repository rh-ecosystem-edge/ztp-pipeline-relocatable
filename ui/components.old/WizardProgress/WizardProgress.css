.wizard-progress {
  margin-top: 2rem;
  margin-left: 25%;
  margin-right: 25%;
}
