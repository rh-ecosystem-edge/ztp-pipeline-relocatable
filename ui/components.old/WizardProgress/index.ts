export * from './WizardProgress';
export * from './WizardProgressContext';
