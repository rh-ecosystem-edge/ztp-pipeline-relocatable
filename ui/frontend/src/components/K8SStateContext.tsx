import React from 'react';
import { isEqual } from 'lodash';

import {
  IpTripletSelectorValidationType,
  K8SStateContextData,
  K8SStateContextDataFields,
  CustomCertsValidationType,
} from './types';
import { ChangeDomainInputType, TlsCertificate } from '../copy-backend-common';
import {
  customCertsValidator,
  domainValidator,
  ipTripletAddressValidator,
  ipWithoutDots,
  passwordValidator,
  usernameValidator,
} from './utils';

const K8SStateContext = React.createContext<K8SStateContextData | null>(null);

export const K8SStateContextProvider: React.FC<{
  children: React.ReactNode;
}> = ({ children }) => {
  const [username, setUsername] = React.useState('');
  const [usernameValidation, setUsernameValidation] = React.useState<string>('');
  const handleSetUsername = React.useCallback(
    (newVal: string) => {
      setUsernameValidation(usernameValidator(newVal));
      setUsername(newVal);
    },
    [setUsername],
  );

  const [password, setPassword] = React.useState('');
  const [passwordValidation, setPasswordValidation] = React.useState(true);
  const handleSetPassword = React.useCallback(
    (newVal: string) => {
      setPasswordValidation(passwordValidator(newVal));
      setPassword(newVal);
    },
    [setPassword],
  );

  const [apiaddr, setApiaddr] = React.useState(ipWithoutDots('192.168.7.200')); // 192168  7200
  const [ingressIp, setIngressIp] = React.useState(ipWithoutDots('192.168.7.201')); // 192168  7201
  const [apiaddrValidation, setApiaddrValidation] = React.useState<IpTripletSelectorValidationType>(
    {
      valid: true,
      triplets: [],
    },
  );
  const handleSetApiaddr = React.useCallback(
    (newIp: string) => {
      setApiaddrValidation(ipTripletAddressValidator(newIp, ingressIp));
      setApiaddr(newIp);
    },
    [setApiaddr, ingressIp],
  );

  const [ingressIpValidation, setIngressIpValidation] =
    React.useState<IpTripletSelectorValidationType>({
      valid: true,
      triplets: [],
    });
  const handleSetIngressIp = React.useCallback(
    (newIp: string) => {
      setIngressIpValidation(ipTripletAddressValidator(newIp, apiaddr));
      setIngressIp(newIp);
    },
    [setIngressIp, apiaddr],
  );

  const [domain, setDomain] = React.useState<string>('');
  const [originalDomain, setOriginalDomain] = React.useState<string>();
  const [domainValidation, setDomainValidation] =
    React.useState<K8SStateContextData['domainValidation']>();
  const handleSetDomain = React.useCallback(
    (newDomain: string) => {
      setDomainValidation(domainValidator(newDomain));
      setDomain(newDomain);
      if (!originalDomain) {
        // Hint: This is expected to be called within initialDataLoad() only
        setOriginalDomain(newDomain);
      }
    },
    [originalDomain],
  );

  const [customCerts, setCustomCerts] = React.useState<ChangeDomainInputType['customCerts']>({});
  const [customCertsValidation, setCustomCertsValidation] =
    React.useState<CustomCertsValidationType>({});

  const setCustomCertificate = React.useCallback(
    (domain: string, certificate: TlsCertificate) => {
      const newCustomCerts = { ...customCerts };
      newCustomCerts[domain] = certificate;
      setCustomCerts(newCustomCerts);
      setCustomCertsValidation(customCertsValidator(customCertsValidation, domain, certificate));
    },
    [customCertsValidation, customCerts, setCustomCerts],
  );

  const isAllValid = React.useCallback(() => {
    const result =
      !usernameValidation &&
      passwordValidation &&
      apiaddrValidation.valid &&
      ingressIpValidation.valid &&
      !domainValidation &&
      !Object.keys(customCertsValidation).find(
        (d) =>
          customCertsValidation[d].certValidated === 'error' ||
          customCertsValidation[d].keyValidated === 'error',
      );
    return result;
  }, [
    apiaddrValidation.valid,
    customCertsValidation,
    domainValidation,
    ingressIpValidation.valid,
    passwordValidation,
    usernameValidation,
  ]);

  const _fv: K8SStateContextDataFields = {
    username,
    password,
    apiaddr,
    ingressIp,
    domain,
    originalDomain,
    customCerts,
  };

  const fieldValues = React.useRef<K8SStateContextDataFields>(_fv);
  fieldValues.current = _fv;

  const [snapshot, setSnapshot] = React.useState<K8SStateContextDataFields>(_fv);
  const setClean = React.useCallback(() => {
    setSnapshot(fieldValues.current);
  }, [fieldValues]);
  const isDirty = React.useCallback((): boolean => {
    return !isEqual(fieldValues.current, snapshot);
  }, [fieldValues, snapshot]);

  const value = {
    ...fieldValues.current,

    isDirty,
    setClean,
    isAllValid,

    usernameValidation,
    handleSetUsername,

    passwordValidation,
    handleSetPassword,

    apiaddrValidation,
    handleSetApiaddr,

    ingressIpValidation,
    handleSetIngressIp,

    domainValidation,
    handleSetDomain,

    customCertsValidation,
    setCustomCertificate,
  };

  return <K8SStateContext.Provider value={value}>{children}</K8SStateContext.Provider>;
};

export const useK8SStateContext = () => {
  const context = React.useContext(K8SStateContext);
  if (!context) {
    throw new Error('useK8SStateContext must be used within K8SStateContextProvider.');
  }
  return context;
};
