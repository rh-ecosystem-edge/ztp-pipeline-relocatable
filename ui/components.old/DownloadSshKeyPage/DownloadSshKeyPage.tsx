import React from 'react';

import { Page } from '../Page';
import { ContentThreeRows } from '../ContentThreeRows';
import { WizardProgress } from '../WizardProgress';
import { useWizardProgressContext } from '../WizardProgress/WizardProgressContext';
import { WizardFooter } from '../WizardFooter';
import { DownloadSshKey } from './DownloadSshKey';

export const DownloadSshKeyPage: React.FC = () => {
  const { setActiveStep } = useWizardProgressContext();
  React.useEffect(() => setActiveStep('sshkey'), [setActiveStep]);
  const [isDownloaded, setDownloaded] = React.useState(false);

  const isNextEnabled = () => isDownloaded;

  return (
    <Page>
      <ContentThreeRows
        top={<WizardProgress />}
        middle={<DownloadSshKey setDownloaded={setDownloaded} />}
        bottom={<WizardFooter back="domain" next="persist" isNextEnabled={isNextEnabled} />}
      />
    </Page>
  );
};
