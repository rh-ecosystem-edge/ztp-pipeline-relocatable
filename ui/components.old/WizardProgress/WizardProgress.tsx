import React from 'react';
import { ProgressStepper, ProgressStep } from '@patternfly/react-core';

import { useWizardProgressContext } from './WizardProgressContext';

import './WizardProgress.css';

export const WizardProgress: React.FC = () => {
  const { steps } = useWizardProgressContext();

  return (
    <ProgressStepper isCenterAligned className="wizard-progress">
      <ProgressStep
        // description=""
        id="username"
        titleId="username-title"
        aria-label="Username step"
        {...steps.username}
      >
        Username
      </ProgressStep>
      <ProgressStep
        id="password"
        titleId="password-title"
        aria-label="Password step"
        {...steps.password}
      >
        Password
      </ProgressStep>
      <ProgressStep id="apiaddr" titleId="apiaddr-title" aria-label="API step" {...steps.apiaddr}>
        API
      </ProgressStep>
      <ProgressStep
        id="ingressip"
        titleId="ingressip-title"
        aria-label="Ingress step"
        {...steps.ingressip}
      >
        Ingress
      </ProgressStep>
      <ProgressStep id="domain" titleId="domain-title" aria-label="Domain step" {...steps.domain}>
        Domain
      </ProgressStep>
      <ProgressStep
        id="sshkey"
        titleId="sshkey-title"
        aria-label="Download SSH key step"
        {...steps.sshkey}
      >
        SSH
      </ProgressStep>
    </ProgressStepper>
  );
};
