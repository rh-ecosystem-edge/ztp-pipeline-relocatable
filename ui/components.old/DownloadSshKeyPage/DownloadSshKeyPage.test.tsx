import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import fetch from 'jest-fetch-mock';

import { DownloadSshKeyPage } from './DownloadSshKeyPage';
import { K8SStateContextProvider } from '../K8SStateContext';
import { WizardProgressContextProvider } from '../WizardProgress';
import { delay } from '../../test-utils';

const Component = () => {
  return (
    <K8SStateContextProvider>
      <WizardProgressContextProvider>
        <MemoryRouter>
          <DownloadSshKeyPage />
        </MemoryRouter>
      </WizardProgressContextProvider>
    </K8SStateContextProvider>
  );
};

describe('DownloadSshKeyPage', () => {
  beforeEach(() => {
    fetch.resetMocks();
  });

  it('can render', async () => {
    fetch.mockResponseOnce(JSON.stringify({ data: { 'invalid.key': 'aaa' } }));
    let container;
    // eslint-disable-next-line testing-library/no-unnecessary-act
    await act(async () => {
      const { container: c } = render(<Component />);
      container = c;
      await delay(1000);
    });

    expect(screen.getByTestId('wizard-footer-button-next')).toBeDisabled();
    expect(screen.getByTestId('wizard-footer-button-next')).toHaveTextContent('Finish setup');
    expect(screen.getByTestId('button-download-ssh-key')).toBeDisabled();
    expect(container).toMatchSnapshot();
  });

  it('can download', async () => {
    // a very simplified stub of a Secret resource...
    fetch.mockResponseOnce(
      JSON.stringify({ data: { 'id_rsa.key': 'aaa' /* Should be Base64 */ } }),
    );
    // eslint-disable-next-line testing-library/no-unnecessary-act
    await act(async () => {
      render(<Component />);
      await delay(1000);
    });

    expect(fetch).toHaveBeenCalledTimes(1);

    expect(screen.getByTestId('wizard-footer-button-next')).toBeDisabled();
    expect(screen.getByTestId('button-download-ssh-key')).not.toBeDisabled();
    screen.getByTestId('button-download-ssh-key').click();
    expect(screen.getByTestId('wizard-footer-button-next')).not.toBeDisabled();
  });
});
