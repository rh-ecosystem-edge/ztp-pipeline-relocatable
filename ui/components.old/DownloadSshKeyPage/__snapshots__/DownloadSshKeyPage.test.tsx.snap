// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadSshKeyPage can render 1`] = `
<div>
  <div
    class="pf-c-background-image"
    style="--pf-c-background-image--BackgroundImage: url(/background.png); --pf-c-background-image--BackgroundImage-2x: url(/background.png); --pf-c-background-image--BackgroundImage--sm: url(/background.png); --pf-c-background-image--BackgroundImage--sm-2x: url(/background.png); --pf-c-background-image--BackgroundImage--lg: url(/background.png); --pf-c-background-image--Filter: url(#patternfly-background-image-filter-overlay0);"
  >
    <svg
      class="pf-c-background-image__filter"
      height="0"
      width="0"
      xmlns="http://www.w3.org/2000/svg"
    >
      <filter
        id="patternfly-background-image-filter-overlay0"
      >
        <fecolormatrix
          type="matrix"
          values="1 0 0 0 0 1 0 0 0 0 1 0 0 0 0 0 0 0 1 0"
        />
        <fecomponenttransfer
          color-interpolation-filters="sRGB"
          result="duotone"
        >
          <fefuncr
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefuncg
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefuncb
            tableValues="0.086274509803922 0.43921568627451"
            type="table"
          />
          <fefunca
            tableValues="0 1"
            type="table"
          />
        </fecomponenttransfer>
      </filter>
    </svg>
  </div>
  <header
    class="pf-c-page__header page-header"
  >
    <div
      class="pf-c-page__header-brand"
    >
      <a
        class="pf-c-page__header-brand-link"
      >
        <img
          alt="Logo"
          src="RedHatLogo.svg"
        />
      </a>
    </div>
    <div
      class="pf-c-page__header-nav"
    >
      <div
        class="page-header-middle"
      >
        Edge cluster setup
      </div>
    </div>
    <div
      class="pf-c-page__header-tools"
    >
      <a
        aria-disabled="false"
        class="pf-c-button pf-m-tertiary"
        data-ouia-component-id="OUIA-Generated-Button-tertiary-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
      >
        Log out
      </a>
    </div>
  </header>
  <div
    class="page-container"
  >
    <div
      class="page-content"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <div
          class="pf-l-stack__item content-three-rows__top"
        >
          <ol
            class="pf-c-progress-stepper pf-m-center wizard-progress"
          >
            <li
              aria-label="Username step"
              class="pf-c-progress-stepper__step pf-m-success"
              id="username"
            >
              <div
                class="pf-c-progress-stepper__step-connector"
              >
                <span
                  class="pf-c-progress-stepper__step-icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="pf-c-progress-stepper__step-main"
              >
                <div
                  aria-labelledby="username username-title"
                  class="pf-c-progress-stepper__step-title"
                  id="username-title"
                >
                  Username
                </div>
              </div>
            </li>
            <li
              aria-label="Password step"
              class="pf-c-progress-stepper__step pf-m-success"
              id="password"
            >
              <div
                class="pf-c-progress-stepper__step-connector"
              >
                <span
                  class="pf-c-progress-stepper__step-icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="pf-c-progress-stepper__step-main"
              >
                <div
                  aria-labelledby="password password-title"
                  class="pf-c-progress-stepper__step-title"
                  id="password-title"
                >
                  Password
                </div>
              </div>
            </li>
            <li
              aria-label="API step"
              class="pf-c-progress-stepper__step pf-m-success"
              id="apiaddr"
            >
              <div
                class="pf-c-progress-stepper__step-connector"
              >
                <span
                  class="pf-c-progress-stepper__step-icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="pf-c-progress-stepper__step-main"
              >
                <div
                  aria-labelledby="apiaddr apiaddr-title"
                  class="pf-c-progress-stepper__step-title"
                  id="apiaddr-title"
                >
                  API
                </div>
              </div>
            </li>
            <li
              aria-label="Ingress step"
              class="pf-c-progress-stepper__step pf-m-success"
              id="ingressip"
            >
              <div
                class="pf-c-progress-stepper__step-connector"
              >
                <span
                  class="pf-c-progress-stepper__step-icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="pf-c-progress-stepper__step-main"
              >
                <div
                  aria-labelledby="ingressip ingressip-title"
                  class="pf-c-progress-stepper__step-title"
                  id="ingressip-title"
                >
                  Ingress
                </div>
              </div>
            </li>
            <li
              aria-label="Domain step"
              class="pf-c-progress-stepper__step pf-m-success"
              id="domain"
            >
              <div
                class="pf-c-progress-stepper__step-connector"
              >
                <span
                  class="pf-c-progress-stepper__step-icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="pf-c-progress-stepper__step-main"
              >
                <div
                  aria-labelledby="domain domain-title"
                  class="pf-c-progress-stepper__step-title"
                  id="domain-title"
                >
                  Domain
                </div>
              </div>
            </li>
            <li
              aria-label="Download SSH key step"
              class="pf-c-progress-stepper__step pf-m-info pf-m-current"
              id="sshkey"
            >
              <div
                class="pf-c-progress-stepper__step-connector"
              >
                <span
                  class="pf-c-progress-stepper__step-icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M512.1,0 C229.7,0 0,229.8 0,512 C0,794.3 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0 Z M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.7,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 Z M512,224 C671.1,224 800,352.9 800,512 C800,671.1 671.1,800 512,800 C352.9,800 224,671.1 224,512 C224,352.9 352.9,224 512,224"
                    />
                  </svg>
                </span>
              </div>
              <div
                class="pf-c-progress-stepper__step-main"
              >
                <div
                  aria-labelledby="sshkey sshkey-title"
                  class="pf-c-progress-stepper__step-title"
                  id="sshkey-title"
                >
                  SSH
                </div>
              </div>
            </li>
          </ol>
        </div>
        <div
          class="pf-l-stack__item pf-m-fill content-three-rows__middle"
        >
          <div
            class="pf-l-stack pf-m-gutter wizard-content"
          >
            <div
              class="pf-l-stack__item"
            >
              <div
                aria-label="Warning Alert"
                class="pf-c-alert pf-m-inline pf-m-warning"
                data-ouia-component-id="OUIA-Generated-Alert-warning-1"
                data-ouia-component-type="PF4/Alert"
                data-ouia-safe="true"
              >
                <div
                  class="pf-c-alert__icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 576 512"
                    width="1em"
                  >
                    <path
                      d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                    />
                  </svg>
                </div>
                <h4
                  class="pf-c-alert__title"
                >
                  <span
                    class="pf-u-screen-reader"
                  >
                    Warning alert:
                  </span>
                  Edge cluster does not store the private key. Please download and keep your key in a safe place.
                </h4>
                
              </div>
              <div
                aria-label="Danger Alert"
                class="pf-c-alert pf-m-inline pf-m-danger download-item__error"
                data-ouia-component-id="OUIA-Generated-Alert-danger-1"
                data-ouia-component-type="PF4/Alert"
                data-ouia-safe="true"
              >
                <div
                  class="pf-c-alert__icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                    />
                  </svg>
                </div>
                <h4
                  class="pf-c-alert__title"
                >
                  <span
                    class="pf-u-screen-reader"
                  >
                    Danger alert:
                  </span>
                  Incorrect SSH key secret
                </h4>
                <div
                  class="pf-c-alert__description"
                >
                  The secret with ssh private key does not meet the required structure.
                   
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-link pf-m-inline"
                    data-ouia-component-id="OUIA-Generated-Button-link-3"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    Try again
                  </button>
                </div>
              </div>
            </div>
            <div
              class="pf-l-stack__item"
            >
              <h1
                class="pf-c-title pf-m-2xl"
                data-ouia-component-id="OUIA-Generated-Title-1"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Download your private SSH key
              </h1>
            </div>
            <div
              class="pf-l-stack__item wizard-sublabel"
            >
              Your account will be granted access to all the nodes.
            </div>
            <div
              class="pf-l-stack__item download-item"
            >
              <div
                class="pf-l-grid"
              >
                <div
                  class="pf-l-grid__item pf-m-2-col pf-m-2-row download-item__icon"
                >
                  <img
                    alt="Book icon"
                    class="download-ssh__book"
                    src="book.svg"
                  />
                </div>
                <div
                  class="pf-l-grid__item pf-m-10-col"
                >
                  <div
                    class="download-item__text"
                  >
                    Edge Cluster Private SSH Key
                  </div>
                </div>
                <div
                  class="pf-l-grid__item pf-m-10-col"
                >
                  <button
                    aria-disabled="true"
                    class="pf-c-button pf-m-link pf-m-disabled"
                    data-ouia-component-id="OUIA-Generated-Button-link-1"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    data-testid="button-download-ssh-key"
                    disabled=""
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                      />
                    </svg>
                     Download
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="pf-l-stack__item content-three-rows__bottom"
        >
          <div
            class="wizard-footer"
          >
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-link"
              data-ouia-component-id="OUIA-Generated-Button-link-2"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              data-testid="wizard-footer-button-back"
              type="button"
            >
              Go back
            </button>
            <button
              aria-disabled="true"
              class="pf-c-button pf-m-primary pf-m-disabled"
              data-ouia-component-id="OUIA-Generated-Button-primary-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              data-testid="wizard-footer-button-next"
              disabled=""
              type="button"
            >
              Finish setup
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
